### ISSUE TITLE
assertion errors

### ISSUE THREAD

**Issue by VikramTiwari** [metabase/metabase#7187]
- Your databases: BigQuery
- Metabase version: 0.28.2
- Metabase internal database: MySQL

Error Logs:
```
 03-23 03:37:05 WARN metabase.query-processor :: {:status :failed,
  :class java.lang.AssertionError,
  :error "Assert failed: (seq dataset-id)",
  :stacktrace
  ["driver.bigquery$mbql__GT_native.invokeStatic(bigquery.clj:342)"
   "driver.bigquery$mbql__GT_native.invoke(bigquery.clj:342)"
   "driver$fn__26446$G__26326__26453.invoke(driver.clj:51)"
   "query_processor.middleware.mbql_to_native$query__GT_native_form.invokeStatic(mbql_to_native.clj:17)"
   "query_processor.middleware.mbql_to_native$query__GT_native_form.invoke(mbql_to_native.clj:12)"
   "query_processor.middleware.mbql_to_native$mbql__GT_native$fn__30803.invoke(mbql_to_native.clj:26)"
   "query_processor.middleware.annotate_and_sort$annotate_and_sort$fn__29197.invoke(annotate_and_sort.clj:41)"
   "query_processor.middleware.limit$limit$fn__30758.invoke(limit.clj:15)"
   "query_processor.middleware.cumulative_aggregations$cumulative_aggregation$fn__30616.invoke(cumulative_aggregations.clj:58)"
   "query_processor.middleware.cumulative_aggregations$cumulative_aggregation$fn__30616.invoke(cumulative_aggregations.clj:58)"
   "query_processor.middleware.format_rows$format_rows$fn__30748.invoke(format_rows.clj:26)"
   "query_processor.middleware.binning$update_binning_strategy$fn__29276.invoke(binning.clj:172)"
   "query_processor.middleware.results_metadata$record_and_return_metadata_BANG_$fn__31975.invoke(results_metadata.clj:89)"
   "query_processor.middleware.resolve$resolve_middleware$fn__28805.invoke(resolve.clj:424)"
   "query_processor.middleware.expand$expand_middleware$fn__30505.invoke(expand.clj:601)"
   "query_processor.middleware.add_row_count_and_status$add_row_count_and_status$fn__28889.invoke(add_row_count_and_status.clj:14)"
   "query_processor.middleware.driver_specific$process_query_in_context$fn__30636.invoke(driver_specific.clj:12)"
   "query_processor.middleware.resolve_driver$resolve_driver$fn__31986.invoke(resolve_driver.clj:15)"
   "query_processor.middleware.cache$maybe_return_cached_results$fn__29357.invoke(cache.clj:146)"
   "query_processor.middleware.catch_exceptions$catch_exceptions$fn__30553.invoke(catch_exceptions.clj:58)"
   "query_processor$process_query.invokeStatic(query_processor.clj:130)"
   "query_processor$process_query.invoke(query_processor.clj:126)"
   "query_processor$run_and_save_query_BANG_.invokeStatic(query_processor.clj:241)"
   "query_processor$run_and_save_query_BANG_.invoke(query_processor.clj:235)"
   "query_processor$fn__32020$process_query_and_save_execution_BANG___32025$fn__32026.invoke(query_processor.clj:281)"
   "query_processor$fn__32020$process_query_and_save_execution_BANG___32025.invoke(query_processor.clj:267)"
   "query_processor$fn__32044$process_query_and_save_with_max_BANG___32049$fn__32050.invoke(query_processor.clj:302)"
   "query_processor$fn__32044$process_query_and_save_with_max_BANG___32049.invoke(query_processor.clj:298)"
   "api.dataset$fn__47791$fn__47794$fn__47795.invoke(dataset.clj:49)"
   "api.common$fn__20318$invoke_thunk_with_keepalive__20323$fn__20324$fn__20325.invoke(common.clj:403)"],
  :query
  {:type "query",
   :query {:source_table 387, :aggregation [["METRIC" 192]]},
   :parameters [],
   :constraints {:max-results 10000, :max-results-bare-rows 2000},
   :info
   {:executed-by 11,
    :context :ad-hoc,
    :card-id nil,
    :nested? false,
    :query-hash [-81, -61, -88, 14, -70, 102, 73, -36, -83, 68, -85, 5, 34, -15, 16, -17, -117, 93, -56, -11, 13, 4, 78, 79, 18, 4, -66, -15, -40, 99, -82, -92],
    :query-type "MBQL"}},
  :expanded-query nil}
```

```
 03-23 03:37:05 WARN metabase.query-processor :: Query failure: Assert failed: (seq dataset-id)
 ["query_processor$assert_query_status_successful.invokeStatic(query_processor.clj:209)"
  "query_processor$assert_query_status_successful.invoke(query_processor.clj:202)"
  "query_processor$run_and_save_query_BANG_.invokeStatic(query_processor.clj:242)"
  "query_processor$run_and_save_query_BANG_.invoke(query_processor.clj:235)"
  "query_processor$fn__32020$process_query_and_save_execution_BANG___32025$fn__32026.invoke(query_processor.clj:281)"
  "query_processor$fn__32020$process_query_and_save_execution_BANG___32025.invoke(query_processor.clj:267)"
  "query_processor$fn__32044$process_query_and_save_with_max_BANG___32049$fn__32050.invoke(query_processor.clj:302)"
  "query_processor$fn__32044$process_query_and_save_with_max_BANG___32049.invoke(query_processor.clj:298)"
  "api.dataset$fn__47791$fn__47794$fn__47795.invoke(dataset.clj:49)"
  "api.common$fn__20318$invoke_thunk_with_keepalive__20323$fn__20324$fn__20325.invoke(common.clj:403)"]
```

**Comment by VikramTiwari** [comment]
Ah! My bad. More details:

It used to happen a lot when going to `dashboard` in versions where going to `/dashboard` went through all the questions. Now it happens to a very few time on some specific questions. There's nothing specific in those questions though.

I will update with more details.

**Comment by flamber** [comment]
A lot has changed since, the new BigQuery driver available in 0.41 likely fixes it - if it wasn't fixed long ago already.
